### ISSUE TITLE
Uploading of files using IFormFile will buffer if over 64kb. Does the file get cleaned up?

### ISSUE THREAD

**Issue by kirk-quinbar** [aspnet/Mvc#7973]
There is an walkthrough of the two main ways of uploading files in asp.net core mvc at the link below. There is a note that if using IFormFile, the file will be buffered to disk if it is over 64Kb. Does this file get cleaned up immediately or is it treated like a temporary file that sits until some process cleans it up?

https://docs.microsoft.com/en-us/aspnet/core/mvc/models/file-uploads?view=aspnetcore-2.1

Any single buffered file exceeding 64KB will be moved from RAM to a temp file on disk on the server. The resources (disk, RAM) used by file uploads depend on the number and size of concurrent file uploads. Streaming isn't so much about perf, it's about scale. If you try to buffer too many uploads, your site will crash when it runs out of memory or disk space.

**Comment by pranavkm** [comment]
cc @Tratcher

**Comment by Tratcher** [comment]
Yes, the file is marked as DeleteOnClose and closed at the end of the request:
https://github.com/aspnet/HttpAbstractions/blob/6ffac0c722b1d71837d438f22da58aefddd6caac/src/Microsoft.AspNetCore.WebUtilities/FileBufferingReadStream.cs#L197
https://github.com/aspnet/HttpAbstractions/blob/87cd79d6fc54bb4abf07c1e380cd7a9498a78612/src/Microsoft.AspNetCore.Http/Internal/BufferingHelper.cs#L53
